adapters/Unity/UnityAdapter/GADMAdapterUnity.m

```diff
@@ -123,68 +130,66 @@ - (NSString *)getPlacementID {
 
 #pragma mark - Unity Delegate Methods
 
-- (void)unityAdsPlacementStateChanged:(NSString *)placementId
+- (void)unityAdsPlacementStateChanged:(NSString *)placementID
                              oldState:(UnityAdsPlacementState)oldState
                              newState:(UnityAdsPlacementState)newState {
-  // This callback is not forwarded to the adapter by the GADMAdapterUnitySingleton and the adapter
-  // should use the unityAdsReady: and unityAdsDidError: callbacks to forward Unity Ads SDK state to
-  // Google Mobile Ads SDK.
+  id<GADMAdNetworkConnector> strongNetworkConnector = _networkConnector;
+  if (strongNetworkConnector && [placementID isEqualToString:_placementID]) {
+    if (newState == kUnityAdsPlacementStateNoFill){
+      NSError *errorWithDescription = GADUnityErrorWithDescription(@"NO_FILL");
+      [strongNetworkConnector adapter:self didFailAd:errorWithDescription];
+    } else if (newState == kUnityAdsPlacementStateDisabled) {
+      NSError *errorWithDescription = GADUnityErrorWithDescription(@"PlACEMENT_DISABLED");
+      [strongNetworkConnector adapter:self didFailAd:errorWithDescription];
+    } else if (newState == kUnityAdsPlacementStateNotAvailable) {
+      NSError *errorWithDescription = GADUnityErrorWithDescription(@"PlACEMENT_NOTAVAILABLE");
+      [strongNetworkConnector adapter:self didFailAd:errorWithDescription];
+    }
```

Comment:
We can move `[strongNetworkConnector adapter:self didFailAd:errorWithDescription];` out of the `if()` and `else if()` blocks.

Additionally, can this be turned into a `switch()` statement with a `default` returning a generic error to capture all possible outcomes?

Reply:
Thank you for clarifying.